servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -400,47 +404,46 @@ public void streamToJournalCheckpoint(JournalOutputStream outputStream) throws I
   public void addInodeFromJournal(InodeEntry entry) {
     if (entry instanceof InodeFileEntry) {
       InodeFile file = ((InodeFileEntry) entry).toInodeFile();
-
-      if (file.getParentId() == mCachedInode.getId()) {
-        mCachedInode.addChild(file);
-      } else {
-        InodeDirectory parentDirectory =
-            (InodeDirectory) mInodes.getFirstByField(mIdIndex, file.getParentId());
-        parentDirectory.addChild(file);
-      }
-      // Update indexes.
-      mInodes.add(file);
-      if (file.isPinned()) {
-        mPinnedInodeFileIds.add(file.getId());
-      }
+      addInodeFromJournalInternal(file);
     } else if (entry instanceof InodeDirectoryEntry) {
       InodeDirectory directory = ((InodeDirectoryEntry) entry).toInodeDirectory();
 
       if (directory.getName() == ROOT_INODE_NAME) {
         // This is the root inode. Clear all the state, and set the root.
-        mRoot = directory;
         mInodes.clear();
+        mPinnedInodeFileIds.clear();
+        mRoot = directory;
         mCachedInode = mRoot;
+        mInodes.add(mRoot);
       } else {
-        // Add this directory to its parent.
-        if (directory.getParentId() == mCachedInode.getId()) {
-          mCachedInode.addChild(directory);
-        } else {
-          InodeDirectory parentDirectory =
-              (InodeDirectory) mInodes.getFirstByField(mIdIndex, directory.getParentId());
-          parentDirectory.addChild(directory);
-        }
-      }
-      // Update indexes.
-      mInodes.add(directory);
-      if (directory.isPinned()) {
-        mPinnedInodeFileIds.add(directory.getId());
```

Comment:
In original code, if every time when `dir.addChild(inode)` is called, `dir.setLastModificationTimeMs(inode.getCreationTimeMs())` is also called, then the original code will work, because each time we add a child, its parent's mod time will be updated to its own creation time, and that's the logic of updating mod time when creating path. 

I thought `InodeDirectory#addChild` will update the parent's mod time.